### PR TITLE
Add delete config option

### LINKED ISSUE #2514: Allow user to remove config from storage and go back to auto mode

### PR DESCRIPTION
**Pull request by bramkragten**
Relies on: https://github.com/home-assistant/home-assistant/pull/20198

Closes https://github.com/home-assistant/home-assistant-polymer/issues/2514

### CODE REVIEW

File: src/panels/lovelace/hui-root.ts [hunk 1]
```diff
@@ -513,6 +516,25 @@ class HUIRoot extends hassLocalizeLitMixin(LitElement) {
...
+      this.lovelace!.enableFullEditMode();
+      alert("Copy the content of the RAW editor and save it in a safe place.");
+      return;
+    }
+    if (
+      prompt(
+        "Are you sure you want to delete your entire Lovelace configuration permanently? Enter 'DELETE' to confirm."
+      ) === "DELETE"
+    ) {
+      this.lovelace!.deleteConfig();
```
Comment (balloob) [inline src/panels/lovelace/hui-root.ts]: We should await this so we can tell them deletion failed.

**Comment by balloob** [comment]
Ok to merge after we release 86.

**Comment by balloob** [comment]
So I am wondering if this is something that we really want to offer as an option.

We maybe want to do it as a "hack" where if you empty the raw config editor, we delete the config instead ? 

I also think that we should consider having a temporary override to show the auto-gen mode or something, I dunno.

**Comment by iantrich** [comment]
I like the temporary override idea

**Comment by bramkragten** [comment]
I like the empty raw editor = delete... 
For automated mode toggle, would that replace unused entities and contain a copy function like discussed in https://github.com/home-assistant/home-assistant-polymer/pull/2260 ?

**Comment by balloob** [comment]
I think that we should indeed replace unused entities. We can have unused entities be a new mode, where we show their current entities and a tab with unused entities or something?

**Comment by iantrich** [comment]
That sounds perfect

**Comment by robbiet480** [comment]
What is the status on this, should it and the [main repo PR](https://github.com/home-assistant/home-assistant/pull/20198) be closed?

**Comment by bramkragten** [comment]
> What is the status on this, should it and the [main repo PR](https://github.com/home-assistant/home-assistant/pull/20198) be closed?

Good question... what do we want to do with it?

**Comment by iantrich** [comment]
I think we should remove the explicit button removal and instead revert on save of an empty config as previously discussed.

**Pull request closed without merging**